cloud_formation/lambda/ingest_populate/ingest_queue_upload.py

```diff
@@ -73,21 +79,30 @@ def handler(args, context):
 
         retry = 3
         while retry > 0:
-            resp = queue.send_messages(Entries=batch)
-            sent += len(resp['Successful'])
-
-            if 'Failed' in resp and len(resp['Failed']) > 0:
-                print("Batch failed to enqueue messages")
-                print("Retries left: {}".format(retry))
-                print("Boto3 send_messages response: {}".format(resp))
-                time.sleep(SQS_RETRY_TIMEOUT)
-
-                ids = [f['Id'] for f in resp['Failed']]
-                batch = [b for b in batch if b['Id'] in ids]
+            try:
+                try_again = False
+                resp = queue.send_messages(Entries=batch)
+                if 'Successful' in resp:
+                    sent += len(resp['Successful'])
+
+                if 'Failed' in resp and len(resp['Failed']) > 0:
```

Comment:
This could be `elif`.

Reply:
I think we need two If statements.  When searching results yesterday I saw this:  Both Sucessful and Failed.

{
  u
  'Successful': [
    {
      u
      'MD5OfMessageBody': '1e29bd6f636f18fd7104aa5de3e4e3a8',
      u
      'Id': '0',
      u
      'MessageId': 'bf8e5f0f-cda1-4c13-b572-0464fd9a9ecb'
    },
    {
      u
      'MD5OfMessageBody': 'bf43f3694069b81d87175d29bcb79b3c',
      u
      'Id': '1',
      u
      'MessageId': '3328e368-f33e-473c-be4c-871cbee21766'
    },
    {
      u
      'MD5OfMessageBody': '7e4c5d4493b557baa24c3b15522271d7',
      u
      'Id': '2',
      u
      'MessageId': 'b925c83a-c72b-4005-8e59-7fbe71ec056b'
    },
    {
      u
      'MD5OfMessageBody': '89c70365867c9877b37f15eaa2a928be',
      u
      'Id': '3',
      u
      'MessageId': '378bb59b-dcf5-49be-970d-4e34f3f741ff'
    },
    {
      u
      'MD5OfMessageBody': 'eb9080ea3116d442283213ce00aabe6a',
      u
      'Id': '4',
      u
      'MessageId': 'ba88ddf4-fe38-4575-854c-70f9aa7ee994'
    },
    {
      u
      'MD5OfMessageBody': '68a23e70c0039b148c8ec18fc4f16c99',
      u
      'Id': '5',
      u
      'MessageId': '27304600-4ddf-4198-8aad-358854eb52fe'
    },
    {
      u
      'MD5OfMessageBody': 'c4d08353699aaf4b19beb4282c061cf9',
      u
      'Id': '8',
      u
      'MessageId': '341b5e06-98c9-48ef-8bde-332199d58abe'
    },
    {
      u
      'MD5OfMessageBody': '4e4ec7a9a105a0e02c7e7069da8c4827',
      u
      'Id': '9',
      u
      'MessageId': '56e9750b-ae61-4b7c-b524-ef0ed55b1dcd'
    }
  ],
  u
  'Failed': [
    {
      u
      'Message': '',
      u
      'SenderFault': False,
      u
      'Code': 'InternalError',
      u
      'Id': '6'
    },
    {
      u
      'Message': '',
      u
      'SenderFault': False,
      u
      'Code': 'InternalError',
      u
      'Id': '7'
    }
  ],
  'ResponseMetadata': {
    'RetryAttempts': 0,
    'HTTPStatusCode': 200,
    'RequestId': 'c583e4ec-caa6-5025-90fe-b06c330c9d56',
    'HTTPHeaders': {
      'x-amzn-requestid': 'c583e4ec-caa6-5025-90fe-b06c330c9d56',
      'date': 'Tue, 04 Dec 2018 21:53:51 GMT',
      'content-length': '2096',
      'content-type': 'text/xml'
    }
  }